tembo-cli/src/cmd/top.rs

```diff
@@ -0,0 +1,172 @@
+use crate::cli::context::{get_current_context, Environment, Profile, Target};
+use crate::cli::tembo_config::InstanceSettings;
+use crate::cmd::apply::get_instance_settings;
+use crate::Args;
+use anyhow::anyhow;
+use anyhow::{Context, Result};
+use hyper::header::ACCEPT;
+use reqwest::header::HeaderMap;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fmt;
+use temboclient::models::{instance, instance_event};
+use temboclient::{
+    apis::{
+        configuration::Configuration,
+        instance_api::{create_instance, get_all, get_instance, put_instance},
+    },
+    models::connection_info,
+};
+use tokio::runtime::Runtime;
+use tokio::time::{interval, Duration};
+
+#[derive(Args)]
+pub struct TopCommand {}
+
+//Format to display the response. Will be changed in beautify.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MetricsResponse {
+    pub status: String,
+    pub data: MetricsData,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MetricsData {
+    pub resultType: String,
+    pub result: Vec<MetricResult>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MetricResult {
+    pub metric: HashMap<String, String>,
+    pub value: (u64, String),
+}
+
+async fn get_instance_id(
+    instance_name: &str,
+    config: &Configuration,
+    env: &Environment,
+) -> Result<Option<String>, anyhow::Error> {
+    let v = get_all(config, env.org_id.clone().unwrap().as_str()).await;
+
+    match v {
+        Ok(result) => {
+            let maybe_instance = result
+                .iter()
+                .find(|instance| instance.instance_name == instance_name);
+
+            if let Some(instance) = maybe_instance {
+                return Ok(Some(instance.clone().instance_id));
+            }
+        }
+        Err(error) => eprintln!("Error getting instance: {}", error),
+    };
+    Ok(None)
+}
+
+async fn fetch_metrics_loop(
```

Comment:
Please check for the case of a new instance, trying to get metrics directly after an instance has first been started